keps/sig-storage/20190204-kubernetes-csi-release-process.md

```diff
@@ -0,0 +1,233 @@
+---
+kep-number: 0
+title: kubernetes-csi release process
+authors:
+  - "@pohly"
+owning-sig: sig-storage
+participating-sigs:
+  - sig-testing
+  - sig-release
+reviewers:
+  - @msau42
+approvers:
+  - @msau42
+  - sig-testing: TBD
+  - sig-release: TBD
+editor: @pohly
+creation-date: 2019-02-04
+last-updated: 2019-02-04
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# kubernetes-csi-release-process
+
+## Table of Contents
+
+- [kubernetes-csi-release-process](#kubernetes-csi-release-process)
+    - [Table of Contents](#table-of-contents)
+    - [Summary](#summary)
+    - [Motivation](#motivation)
+        - [Goals](#goals)
+        - [Non-Goals](#non-goals)
+    - [Proposal](#proposal)
+        - [Versionioning](#versionioning)
+            - [Release artifacts](#release-artifacts)
+            - [Release process](#release-process)
+        - [Implementation Details](#implementation-details)
+        - [Risks and Mitigations](#risks-and-mitigations)
+    - [Graduation Criteria](#graduation-criteria)
+    - [Implementation History](#implementation-history)
+    - [Drawbacks](#drawbacks)
+    - [Alternatives](#alternatives)
+    - [Infrastructure Needed](#infrastructure-needed)
+
+
+## Summary
+
+The Kubernetes [Storage
+SIG](https://github.com/kubernetes/community/tree/master/sig-storage)
+maintains a set of components under the
+[kubernetes-csi](https://github.com/kubernetes-csi) GitHub
+organization. Those components are intentionally not part of core
+Kubernetes even though they are maintained by the Kubernetes project.
+
+This document explains how these components are released.
+
+## Motivation
+
+So far, the process for tagging and building components has been
+fairly manual, with some assistance by Travis CI. There has been no
+automatic end-to-end testing.
+
+With CSI support reaching GA in Kubernetes 1.13, it is time to define
+and follow a less work-intensive and more predictable approach.
+
+### Goals
+
+- define what a "kubernetes-csi release" is
+- unit testing as mandatory pre-submit check for all components 
+- E2E testing as mandatory pre-submit check for components that get
+  deployed in a cluster
+- define a release process for each component and a "kubernetes-csi release"
+
+### Non-Goals
+
+- change responsibilities among the kubernetes-csi maintainers - as before, each
+  component will have some main maintainer who is responsible for releasing updates
+  of that component
+- automatically generate release notes
```

Comment:
I would like to standardize release note generation as part of the release process.  Can we add this as a goal but leave it for some "phase 2" implementation?

Reply:
Maybe for now just leave it as a non-goal but add a note that we want to tackle it in the future.  Then later we can update this kep with more details and move it to a goal

Reply:
Okay.